calculate_trigger.py

```diff
@@ -60,6 +101,16 @@ def search_for_fault_location(
             """Is current tb to short for trigger position"""
             if trigger_position > goldenrun_tb_info.at[idtbinfo, "ins_count"]:
                 idx = idx - 1
+                if idx < 0:
+                    new_lifespan = calculate_lifespan_from_start(
+                        filter_lists,
+                        fault_address,
+                        goldenrun_tb_exec,
+                        goldenrun_tb_info,
+                        trigger_occurrences,
+                    )
+                    new_lifespan += 2
```

Comment:
Why increment by 2? This should reduce the lifespan by the difference between the specified lifespan and the number of instructions from start. Note this means that the resulting lifespan can be negative, meaning that the fault is not within start and end.

Reply:
Changed to add diff of lifespan and trigger_address value